packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -350,12 +439,46 @@ export class Canary extends cdk.Resource {
     };
   }
 
-  private createRunConfig(props: CanaryProps): CfnCanary.RunConfigProperty | undefined {
-    if (!props.environmentVariables) {
+  private createRunConfig(props: CanaryProps, schedule: CfnCanary.ScheduleProperty): CfnCanary.RunConfigProperty | undefined {
+    // Cloudformation implementation made TimeoutInSeconds a required field where it should not (see links below).
+    // So here is a workaround to fix https://github.com/aws/aws-cdk/issues/9300 and still follow documented behavior.
+    // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-synthetics-canary-runconfig.html#cfn-synthetics-canary-runconfig-timeoutinseconds
+    // https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-synthetics/issues/31
+
+    const MAX_CANARY_TIMEOUT = 840;
+    const rateDuration = Schedule.expressionToRateDuration(schedule.expression);
+    const canaryTimeout = rateDuration.toSeconds() <= MAX_CANARY_TIMEOUT ? rateDuration.toSeconds() : MAX_CANARY_TIMEOUT;
+
+    return {
+      timeoutInSeconds: props.timeout?.toSeconds() ?? canaryTimeout,
+      activeTracing: props.tracing,
+      environmentVariables: props.environmentVariables,
+      memoryInMb: props.memorySize?.toMebibytes(),
+    };
+  }
+
+  private createVpcConfig(props: VpcConfiguration | undefined): CfnCanary.VPCConfigProperty | undefined {
```

Comment:
This code was basically lifted from `Function`, let me know if it needs refiniment!